app/models/space.rb

```diff
@@ -40,28 +58,45 @@ def self.sort_by(sort_param)
     if sort_param == 'pri'
       sorted = sorted.reorder(:price_hour)
     elsif sort_param == 'rev'
-      sorted = sorted.select('COUNT(reviews.id) AS reviews_count').
-               joins('LEFT JOIN reviews ON spaces.id = reviews.id').
-               group(:id).reorder('reviews_count')
+      sorted = sorted.joins(:reviews).select('spaces.id, count(reviews.id) as number_of_reviews').group('spaces.id').order('number_of_reviews DESC')
```

Comment:
Thanks for noticing my mistake here. I think your new solution isn't yet on point, though.

Take a look at the query generated when I made a search with the new code:

```sql
 SELECT spaces.*,
		-- (...)
		spaces.id,
		count(reviews.id) as number_of_reviews
  FROM "spaces"
 INNER JOIN "reviews"
    ON "reviews"."space_id" = "spaces"."id"
 WHERE -- (...)
 GROUP BY spaces.id 
 ORDER BY distance ASC, number_of_reviews DESC
 LIMIT 5 OFFSET 0
```

Because spaces were previously sorted by distance, `#reorder` must be called instead of `#order` in order to *(pun intended)* override it. The `INNER JOIN` discards spaces with zero reviews, which may be confusing to the user if he sees some of the spaces disappearing after sorting them by number of reviews.

Here's my attempt at solving this.

```ruby
sorted = sorted.joins('LEFT JOIN reviews ON spaces.id = reviews.space_id').
         select('COUNT(reviews.id) AS reviews_counter').
         group(:id).reorder('reviews_counter DESC')
```

Alternatively, this should work as a scope.

```ruby
scope :sort_by_number_of_reviews, lambda {
    joins('LEFT JOIN reviews ON spaces.id = reviews.space_id').
      select('COUNT(reviews.id) AS reviews_counter').
      group(:id).reorder('reviews_counter DESC')
  }
```

The generated query is what we're looking for.

```sql
 SELECT spaces.*,
		-- (...)
		COUNT(reviews.id) AS reviews_counter
   FROM "spaces"
   LEFT JOIN reviews
     ON spaces.id = reviews.space_id
  WHERE -- (...)
  GROUP BY id
  ORDER BY reviews_counter DESC
  LIMIT 5 OFFSET 0
```